File: StoreReview.Angular/src/app/models/review.ts
import { UserModel } from './user';

export class ReviewModel {
    id: number;
    description: string;
    date: Date;
    ratting: number;
    photoUrls: string[];
    user: UserModel;
    replies: ReviewModel[];


    constructor(id: number, description: string, date: Date, ratting: number, user: UserModel, photoUrls?: string[], replies?: ReviewModel[]) {
        this.id = id;
        this.description = description;
        this.date = date;
        this.ratting = ratting;
        this.photoUrls = photoUrls;
        this.user = user;
        this.replies = replies;
    }
}

File: StoreReview.Core/Domain/Review.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace StoreReview.Core.Domain
{
    public abstract class Review: Entity
    {
        public string Description { get; set; }
        public IEnumerable<Review> Replies { get; set; }
        public DateTime Date { get; set; }
        public float? Ratting { get; set; }
        public long UserId { get; set; }
        public long? ReviewId { get; set; }
        public IEnumerable<File> Photos { get; set; }
        public User User { get; set; }
    }
}
